internal/app/referenceserver/impl.go

```diff
@@ -55,14 +56,10 @@ func (s *conformanceServer) Unary(
 		ctx,
 		req.Msg.ResponseDefinition,
 		req.Header(),
+		nil,
```

Comment:
Let's pass `req.Peer().Query` from everywhere, instead of hard-coding to nil. That way if query string params are inadvertently provided for POST endpoints, the reference server can detect that.